README.md

```diff
@@ -30,54 +30,139 @@ Then:
+and that the test will continue to run even if the Redis client is not ready to respond to the request.
+
+One potential caveat introduced by this new execution model, is that to depend on operations against Redis, all following operations
+should be made in the context of the [promise chain](https://javascript.info/promise-chaining). As demonstrated in the examples below, whenever there is a dependency on a Redis
+operation result or return value, the operation, should be wrapped in a promise itself. That way, a user can perform asynchronous interactions
+with Redis in a seemingly synchronous manner.
+
+For instance, if you were to depend on values stored in Redis to perform HTTP calls, those HTTP calls should be made in the context of the Redis promise chain:
 
 ```javascript
-import redis from "k6/x/redis";
-import http from "k6/http";
+// Instantiate a new redis client
+const redisClient = new redis.Client({
+    addrs: new Array('localhost:6379'),
+    password: 'foobar',
+})
+
+export default function() {
+    // Once the SRANDMEMBER operation is succesfull,
+    // it resolves the promise and returns the random
+    // set member value to the caller of the resolve callback.
+    //
+    // The next promise performs the synchronous HTTP call, and
+    // returns a promise to the next operation, which uses the 
+    // passed URL value to store some data in redis.
+    redisClient.srandmember('client_ids')
+        .then((randomID) => {
+            const url = `https://my.url/${randomID}`
+            const res = http.get(url)
+
+            // Do something with the result...
+
+            // return a promise resolving to the URL
+            return url
+        })
+        .then((url) => redisClient.hincrby('k6_crocodile_fetched', url, 1))
+}
+```
+
+## Example test scripts
+
+In this example we demonstrate two scenarios: one load testing a redis instance, another using redis as an external data store used throughout the test itself.
+
+```javascript
+import { check } from 'k6'
+import http from 'k6/http'
+import redis from 'k6/x/redis'
+import exec from 'k6/execution'
 
 export const options = {
-  vus: 10,
-  duration: "1m",
-};
+    scenarios: {
+        redisPerformance: {
+            executor: 'shared-iterations',
+            vus: 10,
+            iterations: 200,
+            exec: 'measureRedisPerformance',
+        },
+        usingRedisData: {
+            executor: 'shared-iterations',
+            vus: 10,
+            iterations: 200,
+            exec: 'measureUsingRedisData',
+        },
+    },
+}
 
-// Prepare and connect a redis client in the init context
-const r = new redis.Client({
-  addrs: new Array("localhost:6379"),
-  password: "foobar",
-});
+// Instantiate a new redis client
+const redisClient = new redis.Client({
+    addrs: new Array('localhost:6379'),
+    password: 'foobar',
```

Comment:
Should we enforce good practices and use an env var here?

Reply:
Good call 🙇🏻